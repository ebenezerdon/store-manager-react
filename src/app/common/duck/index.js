export {
  doFetchCurrentUser,
  doFetchAllUsers,
  doFetchSaleRecord,
  doFetchProducts
} from './operations';
export { default as fetchAllDataReducer } from './reducers';
export { default as constants } from './constants';
export { default as types } from './types';
export { default as actions } from './actions';
