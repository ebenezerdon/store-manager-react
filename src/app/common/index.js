export { fetchAllDataReducer } from './duck';
export { constants } from './duck';
export { EllipsisLoaderComponent } from './loaders';

export {
  doFetchCurrentUser,
  doFetchAllUsers,
  doFetchSaleRecord,
  doFetchProducts
} from './duck';
