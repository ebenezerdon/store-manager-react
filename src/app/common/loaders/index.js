export { default as EllipsisLoaderComponent } from './EllipsisLoaderComponent';
