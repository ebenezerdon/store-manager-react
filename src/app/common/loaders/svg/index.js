export { default as ellipsisLoader } from './EllipsisLoader.svg';