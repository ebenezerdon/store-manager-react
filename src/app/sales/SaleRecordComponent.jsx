import React from 'react';
import lifecycle from 'react-pure-lifecycle';
import { Link } from 'react-router-dom';
import { constants } from '../common';
import '../style/salesrecord.css';

const methods = {
  componentWillMount({
    fetchStoreData,
    userData,
    saleRecord
  }) {
    if (!saleRecord.data) {
      fetchStoreData()
    }
  }
};

const SaleRecordComponent = ({
  saleRecord,
  saleRecordState,
}) => {
  const listSales = () => {
    const { data } = saleRecord;
    return (
      data.map(sale => (
        <tr>
          <td>{sale.created_at}</td>
          <td>{sale.productname}</td>
          <td>{sale.quantity}</td>
          <td>{sale.price}</td>
          <td><Link to={`/user/${sale.attendant_id}`}>{sale.attendant_id}</Link></td>
        </tr>
      ))
    )
  }

  return (
    <>
      <table id="sales-record">
        <tr>
          <th>Date</th>
          <th>Name of Product</th>
          <th>Quantity</th>
          <th>Unit Price</th>
          <th>Attendant Id</th>
        </tr>
        {saleRecord.data &&
          saleRecord.data.map(sale => (
            <tr>
              <td>{sale.created_at}</td>
              <td>{sale.productname}</td>
              <td>{sale.quantity}</td>
              <td>{sale.price}</td>
              <td>
                <Link to={`/user/${sale.attendant_id}`}>{sale.attendant_id}</Link>
              </td>
            </tr>
          ))
        }
      </table>
    </>
  )
}

export default lifecycle(methods)(SaleRecordComponent);
