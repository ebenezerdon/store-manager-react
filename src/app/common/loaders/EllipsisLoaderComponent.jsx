import React from 'react';
import { ellipsisLoader } from './svg';

const EllipsisLoaderComponent = () => {
  return <img className="ellipsis" src={ellipsisLoader} alt="loader" />;
};

export default EllipsisLoaderComponent;
