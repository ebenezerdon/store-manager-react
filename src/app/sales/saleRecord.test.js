import React from 'react';
import { shallow, mount } from 'enzyme';
import { EllipsisLoaderComponent } from '../common/loaders'
import SaleRecordComponent from './SaleRecordComponent';
import {
  mapDispatchToProps,
  mapStateToProps,
} from './SaleRecordContainer';

describe('SaleRecord Container', () => {
  it('should show initial state', () => {
    const initialState = {
      fetchAllData: { cart: { Data: ['Data'] } },
      errorMessage: '',
      checkout: {
        checkoutState: '',
      },
    };
    expect(mapStateToProps(initialState).fetchAllUserState).not.toBe(false);
  });

  it('should dispatch action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchStoreData();
    expect(typeof dispatch.mock.calls[0][0]).toEqual('function');
  });
});

describe('Sale record component', () => {
  it('should render the component', () => {
    const fetchStoreData = jest.fn();
    const saleRecord = {data: []};
    const data = [{
      created_at: 'created_at',
      productname: 'productname',
      quantity: 'quantity',
      price: 'price'
    }];
    const component = shallow(
      <SaleRecordComponent
        {...{ fetchStoreData, saleRecord, data }}
      />,
    );
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
