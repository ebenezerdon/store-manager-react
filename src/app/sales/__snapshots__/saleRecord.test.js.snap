// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sale record component should render the component 1`] = `
<SaleRecordComponent
  data={
    Array [
      Object {
        "created_at": "created_at",
        "price": "price",
        "productname": "productname",
        "quantity": "quantity",
      },
    ]
  }
  fetchStoreData={[MockFunction]}
  saleRecord={
    Object {
      "data": Array [],
    }
  }
/>
`;
